.github/workflows/cypress_tests_workflow.yml

```diff
@@ -0,0 +1,297 @@
+        run: |
+          echo "OSD_VERSION=$(yarn --silent pkg-version)" >> $GITHUB_ENV
+
+      - name: Generate random password
+        run: |
+          CYPRESS_password=$(openssl rand -base64 20)
+          echo "CYPRESS_password=$CYPRESS_password" >> $GITHUB_ENV
+
+      - name: Setup OpenSearch and Dashboards
+        run: |
+          npm i osd-launcher -g
+
+          if [ "${{ matrix.SECURITY }}" = "true" ]; then
+            osd-launcher -os ${{ env.OSD_VERSION }} -osd . -p ${{ env.CYPRESS_password }} --no-plugins --no-build --destination ~/
+          else
+            osd-launcher -os ${{ env.OSD_VERSION }} -osd . --no-security --no-plugins --no-build --destination ~/
+          fi
+
+      - name: Run OpenSearch
+        run: |
+          # Launch and forget
+          /bin/bash -c ~/OpenSearch-v${{ env.OSD_VERSION }}/bin/opensearch &
+
+      - name: Run OSD
+        run: |
+          WORKSPACE_FLAG=$([ "${{ matrix.WORKSPACE }}" = "true" ] && echo "--workspace.enabled=true" || echo "--workspace.enabled=false")
+          DATA_SOURCE_FLAG=$([ "${{ matrix.DATA_SOURCE }}" = "true" ] && echo "--data_source.enabled=true" || echo "--data_source.enabled=false")
+          QUERY_ENHANCEMENTS_FLAG=$([ "${{ matrix.QUERY_ENHANCEMENTS }}" = "true" ] && echo "--uiSettings.overrides["query:enhancements:enabled"]=true" || echo "--uiSettings.overrides["query:enhancements:enabled"]=false")
+          USE_NEW_HOME_PAGE_FLAG=$([ "${{ matrix.USE_NEW_HOME_PAGE }}" = "true" ] && echo "--uiSettings.overrides["home:useNewHomePage"]=true" || echo "--uiSettings.overrides["home:useNewHomePage"]=false")
+
+          # Launch and forget
+          /bin/bash -c "node scripts/opensearch_dashboards --dev --no-base-path --no-watch --savedObjects.maxImportPayloadBytes=10485760 --server.maxPayloadBytes=1759977 --logging.json=false --data.search.aggs.shardDelay.enabled=true $WORKSPACE_FLAG $DATA_SOURCE_FLAG $QUERY_ENHANCEMENTS_FLAG $USE_NEW_HOME_PAGE_FLAG" &
+
+      - name: Wait for 15 seconds
+        run: sleep 15s
+        shell: bash
+
+      - name: Run tests
+        run: yarn test:cypress --spec "${{ matrix.spec_pattern }}"
+
+      - name: Upload coverage
+        id: upload-code-coverage
+        uses: codecov/codecov-action@v5
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          directory: ./coverage
+          flags: "cypress:${{ matrix.run_flag }}"
+
+      - name: Generate safe artifact name
+        if: always()
+        id: safe-name
+        run: |
+          value=$(echo "${{ matrix.run_flag }}" | sed -E 's/[^a-z0-9_-]+/_/gi' | sed -E 's/^_|_$//g')
+          echo "value=${value}" >> $GITHUB_OUTPUT
+
+      - name: Upload test artifacts
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: cypress_artifact_${{ steps.safe-name.outputs.value }}
+          path: |
+            cypress/screenshots
+            cypress/videos
+            coverage
+          retention-days: 1
```

Comment:
can code coverage work with split test buckets? We currently got all of our `query-enhancements` test split into 5 buckets due to how long they take

Reply:
Yes.